x-pack/test/plugin_functional/plugins/resolver_test/public/plugin.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Plugin, CoreSetup } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { IEmbeddable, IEmbeddableStart } from '../../../../../../src/plugins/embeddable/public';
+
+export class ResolverTestPlugin implements Plugin {
+  private resolveEmbeddable!: (
+    value: IEmbeddable | undefined | PromiseLike<IEmbeddable | undefined> | undefined
+  ) => void;
+  private embeddablePromise: Promise<IEmbeddable | undefined> = new Promise<
```

Comment:
Actually, I think this is an anti-pattern.  What you really want is to access embeddable via AppMountContext, which it's not available yet (I just hit this on another PR of mine, adding it now, it should be very quick to add...)

@joshdover  - is this kind of anti-pattern codified anywhere?  I've seen this come up a few times, but I have no where to point people to documentation that says "don't do this", but I'm pretty sure somewhere it says not to.

Reply:
Actually, @oatkiller, if you could add that functionality in this PR we'd know for sure it works since this is a test.  I'm working on a test PR but it won't be quick to merge.  

To add this to mount context, you can follow what search service does:

https://github.com/elastic/kibana/blob/master/src/plugins/data/public/search/search_service.ts#L45

and

https://github.com/elastic/kibana/blob/master/src/plugins/data/public/search/search_service.ts#L81

That should simplify this a ton.

Reply:
It is an anti-pattern of sorts, but we're also about to expose an API that does this automatically without requiring plugins to expose context: #49691

Reply:
@stacey-gammon are you suggesting that the embeddable plugin should expose itself on AppMountContext via something like `core.application.registerMountContext`?

Is it an antipattern because the test plugin depends on embeddable?

Reply:
In the current state of the repo, the embeddable plugin should expose itself on the AppMountContext, which would make your access here much simpler.

However, in https://github.com/elastic/kibana/pull/50231, context will be deprecated (though not yet removed) and you'll be able to use the `core.getStartServices()` API to access all of your plugin dependencies.

Since we're in an in-between state now, I think it's fine if we leave as-is and replace this with the new API once 50231 is merged.